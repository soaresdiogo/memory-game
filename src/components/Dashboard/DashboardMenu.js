import React from 'react';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

const styles = (theme) => ({
  root: {
    padding: theme.spacing(6),
  },
  title: {
    paddingBottom: theme.spacing(6),
  },
});

const DashboardMenu = ({ children, classes, title }) => (
  <Paper data-testid="dashboardMenu" className={classes.root}>
    <Typography
      data-testid="dashboardMenuTitle"
      variant="h4"
      align="center"
      className={classes.title}
    >
      {title}
    </Typography>
    {children}
  </Paper>
);

DashboardMenu.defaultProp = {
  classes: PropTypes.node,
  children: PropTypes.node,
};

DashboardMenu.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withStyles(styles)(DashboardMenu);
