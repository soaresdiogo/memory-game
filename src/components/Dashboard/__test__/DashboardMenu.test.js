import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import DashboardMenu from '../DashboardMenu';

import '@testing-library/jest-dom/extend-expect';

const DashboardMenuComponent = <DashboardMenu title="Memory Game" />;

const renderDashbordMenu = () => {
  const { getByTestId } = render(DashboardMenuComponent);
  const container = getByTestId('dashboardMenu');
  return {
    getByTestId,
    container,
  };
};

describe('Dashboard menu component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(DashboardMenuComponent, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders div correctly', () => {
    const { container, getByTestId } = renderDashbordMenu();
    expect(container.children.length).toBe(1);
    expect(getByTestId('dashboardMenuTitle')).toBeDefined();
    expect(getByTestId('dashboardMenuTitle').textContent).toBe('Memory Game');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(DashboardMenuComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
