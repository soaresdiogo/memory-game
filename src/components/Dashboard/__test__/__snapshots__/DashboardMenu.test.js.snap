// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard menu component matches snapshot 1`] = `
<div
  className="MuiPaper-root DashboardMenu-root-5 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="dashboardMenu"
>
  <h4
    className="MuiTypography-root DashboardMenu-title-6 MuiTypography-h4 MuiTypography-alignCenter"
    data-testid="dashboardMenuTitle"
  >
    Memory Game
  </h4>
</div>
`;
